### PR TITLE
[UI-55] Credit card input

### PR DESCRIPTION
**Pull request by stuniel**
#### Task

- **Issue or task referred:** UI-55
- **Feature or Bugfix:** Feature

I created and integrated a credit card input component. It wraps name, card number, expiration date and cvc inputs in FormHandler component. This package also includes CreditCardNumberInput that handles credit card number and ExpirationDateInput that handles month and year.

#### Checklist

- [ ] Required unit tests prepared
- [ ] Documentation prepared
- [ ] External licenses validated (ideally MIT)

### CODE REVIEW

File: packages/credit-card/styles/theming.sass [hunk 1]
```diff
@@ -0,0 +1,4 @@
+@import './config'
+
+@include module('credit-card')
+  display: block
```
Comment (PiotrGrzechnik) [inline packages/credit-card/styles/theming.sass]: This should be in geometry or should be deleted.

File: packages/fieldset/tests/Fieldset.test.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import React from 'react'
+import { shallow, mount } from 'enzyme'
+
+import Fieldset from '../src/Fieldset'
+import { Icon } from '@talixo/icon'
+
+describe('<Fieldset />', () => {
+  it('renders children correctly', () => {
+    const wrapper = mount(<Fieldset legend='Personal info' asideLegend={<span><Icon name='credit-card' /> Credit Card</span>} />)
```
Comment (PiotrGrzechnik) [inline packages/fieldset/tests/Fieldset.test.js]: After `mount` you should also `unmount` at the end of test.
The same above in each `mount `

File: packages/credit-card/src/CreditCardInput.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+          label={cvcLabel}
+          onBlur={onBlur}
+          onChange={(value) => onChange(value, 'cvc')}
+          onFocus={onFocus}
+          value={values.cvc}
+        >
+          <TextInput
+            autoComplete='cc-csc'
+            maxLength={4}
+            size={4}
```
Comment (maciek15) [inline packages/credit-card/src/CreditCardInput.js]: It allows passing strings to input

File: packages/credit-card/src/ExpirationDateInput.js [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+  className: PropTypes.string,
+
+  /** Handler for onChange event. */
+  onChange: PropTypes.func,
+
+  /** Value to be displayed inside inputs. */
+  value: PropTypes.shape({
+    month: PropTypes.number,
+    year: PropTypes.number
+  })
```
Comment (maciek15) [inline packages/credit-card/src/ExpirationDateInput.js]: It's quite obvious here but you could provide information that this is a expiry month and year

File: packages/credit-card/src/ExpirationDateInput.js [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+}
+
+const defaultProps = {}
+
+// Set minimum year to be the current year.
+const minYear = new Date().getFullYear()
+// Set maximum year to be 40 years from the current year.
+const maxYear = minYear + 40
```
Comment (maciek15) [inline packages/credit-card/src/ExpirationDateInput.js]: I think this is not wrong but you it could also be passed as prop not hard-coded as 40

**Pull request closed without merging**